### DISCUSSION TITLE
[CSV Import] Unable to import object with variants inside

### DISCUSSION THREAD

**Discussion by pento-thal** [pimcore/pimcore#8957] [Q&A / Support] [answered]
I'm trying to setup a clothing catalogue by using a CSV import.
My main goal is to replicate the current parent-child structure by using object and variants in PIMCORE.
The results should be something like this:

![image](https://user-images.githubusercontent.com/83424908/116603849-b332b780-a92d-11eb-83fe-6ec9ad02fa07.png)

So, I would like to setup a CSV file able to import:
-  **main object** (model)
- - **secondary object** (model_color) under the first one
- - - **variants** (model_colors_size)

There is a way to do something like this?

**Comment by pento-thal** [comment]
I'm trying with this csv setup:
![image](https://user-images.githubusercontent.com/83424908/116716847-84bde680-a9d8-11eb-9147-c9bc2e45fa73.png)

With this Resolver Settings:
![image](https://user-images.githubusercontent.com/83424908/116718530-580ace80-a9da-11eb-834b-fde824ed0925.png)

**Reply by ghost** [reply]
@pento-thal  I am facing the same issue.
you can use import object events  https://github.com/pimcore/pimcore/blob/6.9/lib/Event/DataObjectImportEvents.php

**Comment by pento-thal** [comment, marked as answer]
Thanks for the reply, I found the right import setting:
![image](https://user-images.githubusercontent.com/83424908/117483415-f9a79800-af65-11eb-8e0b-974fd2c24d9e.png)
